Tests: Update buildall to test modem drivers

galak opened the pull request.

galak committed:
tests: drivers: build_all: Rework modem test to be more generic

Add support to build all current modem drivers (WNCM14A2A, GSM PPP, and
ublox-sara).  We need a unique config for ublox-sara to be built since
the networking params conflict with other modem drivers.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
drivers: modem: wncm14a2a: fix build if CONFIG_MODEM_SIM_NUMBERS=n

If CONFIG_MODEM_SIM_NUMBERS is 'n' than the data_imei field of struct
modem_context doesn't exist, so add ifdef protection for that case.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

rerickson1 commented on tests/drivers/build_all/modem.conf:
```diff
@@ -2,9 +2,14 @@ CONFIG_TEST=y
 CONFIG_MODEM_HL7800_SET_APN_NAME_ON_STARTUP=y
 CONFIG_MODEM_HL7800_APN_NAME="internet"
 CONFIG_MODEM_HL7800_RAT_M1=y
+CONFIG_MODEM_WNCM14A2A=y
+CONFIG_MODEM_GSM_PPP=y
+CONFIG_MODEM_GSM_UART_NAME="TEST_UART_CTLR"
```
Looks good. Add SARA-R4 modem to test as well?

galak replied:
Was lazy with SARA-R4 since it conflicts with the other modem drivers (because of networking stack).  Will add it as an explicit test.

mike-scott replied:
Is it SARA_R4 that conflicts or is it WNCM14A2A?

mike-scott replied:
Also need to add a test for HL7800

galak replied:
SARA_R4 since network offload is expected from what I can tell.

mike-scott replied:
SARA_R4 uses `NET_OFFLOAD` + `NET_SOCKETS_OFFLOAD`
but WNCM14A2A also uses `NET_OFFLOAD`(no `NET_SOCKETS_OFFLOAD` in this case)

galak committed:
dts: bindings: test: Update vnd,serial to not require interrupts

We aren't using vnd,serial for anything currently.  Remove requiring
interrurpts property as that requirement isn't needed.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

mike-scott commented on drivers/modem/wncm14a2a.c:
```diff
@@ -1485,7 +1485,9 @@ static int wncm14a2a_init(const struct device *dev)
 	ictx.mdm_ctx.data_manufacturer = ictx.mdm_manufacturer;
 	ictx.mdm_ctx.data_model = ictx.mdm_model;
 	ictx.mdm_ctx.data_revision = ictx.mdm_revision;
+#ifdef CONFIG_MODEM_SIM_NUMBERS
```
Do we only hit this in test cases?

galak replied:
seems that way.

rerickson1 commented on tests/drivers/build_all/modem_ublox_sara.conf:
```diff
@@ -0,0 +1,6 @@
+CONFIG_TEST=y
+CONFIG_TEST_RANDOM_GENERATOR=y
+CONFIG_NETWORKING=y
```
This will enable `MODEM_HL7800` for the `pinnacle_100_dvk` board. Either need to add `NET_IPV4` to this config or exclude `pinnacle_100_dvk` from `drivers.modem.ublox_sara.build`.

galak replied:
I just excluded it, seems easier.
